tasks/periodic_actor_dump/miner_actor/tasks.go

```diff
@@ -228,15 +228,15 @@ func calculateTerminateFeeForSector(currentEpoch abi.ChainEpoch, sector *miner.S
 	// How long has the sector been in power? Cap this value.
 	cappedSectorAge := min(currentEpoch-sector.PowerBaseEpoch, lifetimeCap)
 	// expected_reward = ExpectedDayReward * cappedSectorAge
-	expectedReward := big.Mul(sector.ExpectedDayReward, big.NewInt(int64(cappedSectorAge)))
+	expectedReward := big.Mul(*sector.ExpectedDayReward, big.NewInt(int64(cappedSectorAge)))
 	// relevant_replaced_age = min(PowerBaseEpoch - Activation, lifetimeCap - cappedSectorAge)
 	relevantReplacedAge := big.NewInt(int64(min(sector.PowerBaseEpoch-sector.Activation, lifetimeCap-cappedSectorAge)))
 	// Add the replaced sector's contribution.
-	expectedReward = big.Add(expectedReward, big.Mul(sector.ReplacedDayReward, relevantReplacedAge))
+	expectedReward = big.Add(expectedReward, big.Mul(*sector.ReplacedDayReward, relevantReplacedAge))
 	// penalized_reward = expected_reward / TERMINATION_REWARD_FACTOR_DENOM
 	penalizedReward := big.Div(expectedReward, big.NewInt(int64(TERMINATION_REWARD_FACTOR_DENOM)))
 	// Termination fee = ExpectedStoragePledge + (penalized_reward / EPOCHS_IN_DAY)
-	return big.Add(sector.ExpectedStoragePledge, big.Div(penalizedReward, big.NewInt(int64(EPOCHS_IN_DAY))))
+	return big.Add(*sector.ExpectedStoragePledge, big.Div(penalizedReward, big.NewInt(int64(EPOCHS_IN_DAY))))
```

Comment:
beware that these are going to start being `nil` from nv25 so you should be careful to guard for them

But also, maybe this should be replaced with a go-state-types variation of this since termination fees have been "simplified" and have a different calculation now. See https://github.com/filecoin-project/go-state-types/pull/376 for some exposed methods that help with this; https://github.com/filecoin-project/lotus/pull/12968 has use of them but also exposes them through the lotus go-state-types abstractions which might be more helpful. See the itest in there for how it can be used (a couple of variations).